assignment/core/src/main/java/com/assignment/core/schedulers/NodeDeletionScheduler.java

```diff
@@ -0,0 +1,57 @@
+package com.assignment.core.schedulers;
+
+import com.assignment.core.services.NodeDeletionService;
+import org.apache.sling.commons.scheduler.ScheduleOptions;
+import org.apache.sling.commons.scheduler.Scheduler;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Modified;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A scheduler that periodically triggers the deletion of nodes under a specified JCR path.
+ */
+@Component(
+        service = Runnable.class,
+        immediate = true
+)
+public class NodeDeletionScheduler implements Runnable {
+
+    private static final Logger LOG = LoggerFactory.getLogger(NodeDeletionScheduler.class);
+
+    @Reference
+    private Scheduler scheduler;
+
+    @Reference
+    private NodeDeletionService nodeDeletionService;
+
+    private String scheduleExpression = "* * * * *"; // Runs every 3 minutes
+    private String rootPath = "/content/assignment/us/en/my-assignment-page";
+
+    /**
+     * Activates the scheduler and sets it to run according to the specified cron expression.
+     */
+    @Modified
+    public void activate() {
```

Comment:
Use @Activate for activate method not @Modified it is for modified method. Also add deactivate, modified, add and remove method of scheduler too.